src/plots/gl3d/index.js

```diff
@@ -51,6 +51,14 @@ exports.plot = function plotGl3d(gd) {
             scene = sceneLayout._scene;
 
         if(!scene) {
+            if(fullLayout[sceneId].camera.up.z !== 1 &&
+                fullLayout[sceneId].dragmode === 'turntable') {
+                fullLayout[sceneId].dragmode = 'orbit';
```

Comment:
Would be better to do this in [`supplyDefaults`](https://github.com/plotly/plotly.js/blob/3a32ac00cf5e8619932f874960f11b4daf7ceb40/src/plots/gl3d/layout/defaults.js#L113)

Not sure where we ensure `||up||===1` but it doesn't look like that's done in `supplyDefaults`, so I guess at that point we might need a more complicated test (`up.x || up.y || up.z < 0` perhaps?)

Reply:
I like the idea of having more complicated test as you mentioned. On the other hand I was wondering that it is now rather difficult for the users to compute and pass z vector components. Could we add a function to normalize user inputs for this vector before reaching there too?

Reply:
Only the direction of `up` matters, right? So feels to me like we shouldn't require it to be normalized at all. If the user passes in (0,0,0) we should revert to (0,0,1), otherwise seems to me we should normalize it for our own use but leave the un-normalized value in `layout` and `fullLayout`.